### PR TITLE
bitbucket-cli: init at 1.0.0

### PR DESCRIPTION
**Pull request by attilaolah**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
  - I've been using this binary in a (private) flake for some time and decided to upstream it now.
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

NOTE: Upstream tests are are using internal credentials, hence disabled.

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/version-management/bitbucket-cli/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  # Tests seem to be using Swisscom's live servers.
+  doCheck = false;
+
+  meta = with lib; {
+    description = "A Bitbucket Enterprise CLI";
+    homepage = "https://github.com/swisscom/bitbucket-cli";
+    mainProgram = "bin/${pname}";
+    license = licenses.mit;
+    maintainers = with maintainers; [ attila ];
+    platforms = with platforms; linux ++ darwin;
```
Comment (Aleksanaa) [inline pkgs/applications/version-management/bitbucket-cli/default.nix:28]:
```suggestion
  meta = {
    description = "Bitbucket Enterprise CLI";
    homepage = "https://github.com/swisscom/bitbucket-cli";
    mainProgram = "bitbucket-cli";
    license = lib.licenses.mit;
    maintainers = with lib.maintainers; [ attila ];
    platforms = with lib.platforms; linux ++ darwin;
```

File: pkgs/applications/version-management/bitbucket-cli/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  buildGoModule,
+  fetchFromGitHub,
+}:
+buildGoModule rec {
+  pname = "bitbucket-cli";
+  version = "1.0.0";
+
+  src = fetchFromGitHub {
+    owner = "swisscom";
+    repo = pname;
```
Comment (Aleksanaa) [inline pkgs/applications/version-management/bitbucket-cli/default.nix:12]:
```suggestion
    repo = "bitbucket-cli";
```

**Review by Aleksanaa — COMMENTED** [review]
Please split into two commits:

1. maintainers: add attila
2. bitbucket-cli: init at 1.0.0

**Comment by attilaolah** [comment]
Done.

File: maintainers/maintainer-list.nix [hunk 3]
```diff
@@ -2028,6 +2028,14 @@
     github = "Atry";
     githubId = 601530;
   };
+  attila = {
+    name = "Attila Oláh";
+    email = "attila@dorn.haus";
+    matrix = "@user:example.org";
```
Comment (attilaolah) [inline maintainers/maintainer-list.nix:2034]: Oh wait, this is not good.

Comment (attilaolah) [reply maintainers/maintainer-list.nix]: OK I'll just remove this, I don't use Matrix that often anyway.

File: pkgs/applications/version-management/bitbucket-cli/default.nix
Comment (Aleksanaa) [inline pkgs/applications/version-management/bitbucket-cli/default.nix:1]: Oh this should be `pkgs/by-name/bi/bitbucket-cli/package.nix`. It is not registered to top-level now.

Comment (attilaolah) [reply pkgs/applications/version-management/bitbucket-cli/default.nix:1]: OK done. Sorry I don't really understand the directory structure here, I'll read up on that.

Comment (attilaolah) [reply pkgs/applications/version-management/bitbucket-cli/default.nix:1]: Ah and it actually builds now. This explains the troubles I was having while trying to build it earlier.

**Pull request merged**